webrtc/README.md

```diff
@@ -260,21 +262,50 @@ be the same. On mismatch the final Noise handshake MUST fail.
 
 ## Multiplexing
 
-After [Connection Security](#connection-security):
-
-1. On success of the authentication handshake _X_, the used datachannel is
-   closed and the plain WebRTC connection is used with its multiplexing
-   capabilities via datachannels.
-
-### Open Questions
-
-- Can we use WebRTC’s data channels in _Browser_ to multiplex a single
-  connection, or do we need to run an additional multiplexer (e.g. yamux) on top
-  of a WebRTC connection and WebRTC datachannel? In other words, does WebRTC
-  provide all functionality of a libp2p muxer like Yamux (e.g. flow control)?
-
-  Yes, with WebRTC's datachannels running on top of SCTP, there is no need for
-  additional multiplexing.
+Following [Connection Security](#connection-security).
+
+The WebRTC browser APIs do not support half-closing nor resets of streams.
+[`RTCDataChannel.close()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/close)
+flushes the remaining messages and closes the local write and read side. After
+calling `RTCDataChannel.close()` one can no longer read from the channel. This
+lack of functionality is problematic, given that libp2p protocols running on top
+of transport protocols, like WebRTC, expect to be able to half-close or reset a
+stream. See [Connection Establishment in
+libp2p](https://github.com/libp2p/specs/blob/master/connections/README.md#definitions).
+
+To support half-closing and resets of streams, libp2p WebRTC uses message
+framing. Messages on a `RTCDataChannel` are embedded into the Protobuf message
+below and send on the `RTCDataChannel` prefixed with the message length in
+bytes, encoded as an unsigned variable length integer as defined by the
+[multiformats unsigned-varint spec][uvarint-spec].
+
+``` proto
+syntax = "proto2";
+
+package webrtc.pb;
+
+message Message {
+  enum Flag {
+    // The sender will no longer send messages.
+    CLOSE_WRITE = 0;
```

Comment:
```suggestion
    FIN = 0;
```

Reply:
Done in https://github.com/mxinden/specs/pull/1/commits/93df7e397e7ba79962ed7bdfbf6d4da40985011c.